### PR TITLE
Add Bessel function of the second kind of order one, bessely1

### LINKED ISSUE #31: RFC: add `y1`, the Bessel function of the second kind of order 1
## Checklist

> Please ensure the following tasks are completed before filing an issue.

* [x] Read and understood the [Code of Conduct][code-of-conduct].
* [x] Searched for existing issues and pull requests.
* [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
* [x] If this is a feature request, the issue name begins with `RFC: `.

## Description

> Description of the issue (or feature request).

Bessel functions find frequent application in areas of scientific computing. This RFC proposes to add the [Bessel](https://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.special.y1.html) function of the second kind of order `1`. Package name:  `@stdlib/math/base/special/y1`.

## Related Issues

> Does this issue (or feature request) have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this issue (or feature request)? This may include screenshots, references, stack traces, sample output, and/or implementation notes.

Reference implementations:

* [SciPy](https://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.special.y1.html)

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md
[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by rreusser**
Resolves #31.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds `bessely1` with analogous caveats on subnormal accuracy.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/bessely1/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+*
+* @module @stdlib/math/base/special/bessely1
+*
+* @example
+* var PINF = require( '@stdlib/math/constants/float64-pinf' );
+* var NINF = require( '@stdlib/math/constants/float64-ninf' );
+* var y1 = require( '@stdlib/math/base/special/bessely1' );
+*
+* var v = y1( 0.0 );
+* // returns -Infinity
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bessely1/lib/index.js:14]: This value does not match the REPL text.

File: lib/node_modules/@stdlib/math/base/special/bessely1/docs/repl.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    Returns
+    -------
+    y: number
+        Function value.
+
+    Examples
+    --------
+    > var y = {{alias}}( 0.0 )
+    1.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bessely1/docs/repl.txt]: This does not match the README.

File: lib/node_modules/@stdlib/math/base/special/bessely1/lib/y1.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+// MAIN //
+
+/**
+* Computes the Bessel function of the second kind of order zero.
+*
+* ## Notes
+*
+* -   Accuracy for subnormal `x` is very poor. Full accuracy is achieved at 1.0e-308 but trends progressively to zero at 5e-324. This suggests that underflow (or overflow, perhaps due to a reciprocal) is effectively cutting off digits of precision until the computation loses all accuracy at 5e-324.
+*
+* @private
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bessely1/lib/y1.js]: This should not have the `private` annotation. This is the main exported function, correct?

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/bessely1/lib/y1.js]: I removed from `besselj0` and `besselj1`.

File: lib/node_modules/@stdlib/math/base/special/bessely1/lib/y1.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+	}
+	y = 8.0 / x;
+	y2 = y * y;
+	rc = polyC( y2 );
+	rs = polyS( y2 );
+	f = ONE_DIV_SQRT_PI / sqrt( x );
+	/*
+	* This code is really just:
+	*
+	* z = x - 0.75 * PI;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bessely1/lib/y1.js]: Missing code fences.

File: lib/node_modules/@stdlib/math/base/special/bessely1/test/test.js [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+	t.end();
+});
+
+tape( 'the function returns `NaN` for negative numbers', function test( t ) {
+	var v;
+	var x;
+	var i;
+
+	for ( i = 0; i < 1000; i++ ) {
+		x = -randu() * 100.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/bessely1/test/test.js]: `randu()` can return `0`. Subtract `float64-eps` to ensure never `0`.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/math/base/special/bessely1/docs/repl.txt (hunk 2 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/bessely1/docs/repl.txt]: Good catch. Fixed.

File: lib/node_modules/@stdlib/math/base/special/bessely1/test/test.js (hunk 5 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/bessely1/test/test.js]: Nice catch!

**Comment by rreusser** [comment]
`bessely0` also had all the same issues. I've fixed them in both.

**Comment by kgryte** [comment]
Thanks!

File: lib/node_modules/@stdlib/math/base/special/bessely1/lib/index.js (hunk 1 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/bessely1/lib/index.js:14]: This one is correct. The other has been changed.

**Review by kgryte — APPROVED** [review]

**Pull request merged**